lib/partner/client.rb

```diff
@@ -0,0 +1,14 @@
+module Partner
+  class Client
+    API_KEY = "secretpartnerkey".freeze
```

Comment:
That key was just for testing and will be deleted. You should probably use `ENV["PARTNER_KEY"]`

Reply:
yeah, this will be refactored, just did the initial working one.

Reply:
I was thinking about this client and I think it might make more sense to just add this into the client service we already have: `https://github.com/rubyforgood/diaper/blob/master/app/services/diaper_partner_client.rb`

Maybe something like:
```
  def self.get(attributes)
    return if Rails.env != "production"

    partner = { partner:
                    {
                        diaper_partner_id: attributes["id"]
                    } }

    uri = URI(ENV["PARTNER_REGISTER_URL"] + "/#{attributes["id"]}")
    req = Net::HTTP::Get.new(uri, "Content-Type" => "application/json")
    req.body = partner.to_json
    req["Content-Type"] = "application/json"
    req["X-Api-Key"] = ENV["PARTNER_KEY"]

    response = https(uri).request(req)

    response.body
  end
```

You'll probably need to add in a `JSON.parse` on output. It will save us having to add in a rest-client gem and stay consistent with how we're currently interacting with the partner system.